src/eval/operation.rs

```diff
@@ -2916,12 +3010,15 @@ impl RecordDataExt for RecordData {
 mod tests {
     use super::*;
     use crate::cache::resolvers::DummyResolver;
+    use crate::eval::cache::CBNCache;
     use crate::eval::Environment;
 
+    type EC = CBNCache;
```

Comment:
Same comment here re: the type alias perhaps being unnecessary.

Reply:
We left these around in case we implemented a different cache in order to have tests work for both implementations. But I'm sure there's still too much `Thunk` leakage to have this work without major fixes

Reply:
That's fair - it's totally up to you whether you want to keep them in. Personally I tend to err on the side of removing any code that isn't strictly necessary to the current change, as I don't think it'd necessarily be clear to future engineers what the intention was here, but I don't think it's a huge issue to leave it as-is.

Reply:
I'll leave these around for now